src/main/java/emissary/core/IBaseDataObjectHelper.java

```diff
@@ -219,6 +220,26 @@ public static void addParentInformationToChildren(final IBaseDataObject parent,
         }
     }
 
+    /**
+     * Used to propagate needed parent metadata parameter information to a sprouted child except for those parameters whose
+     * names are specified in a regular expression pattern.
+     *
+     * @param parentIBaseDataObject the source of parameters to be copied
+     * @param childIBaseDataObject the destination for parameters to be copied
+     * @param excludedParametersPattern compiled regular expression specifying parameter names not to copy down
+     */
+
+    public static void addParentInformationToChildExcluding(final IBaseDataObject parentIBaseDataObject,
+            final IBaseDataObject childIBaseDataObject,
+            final Pattern excludedParametersPattern) {
+        Validate.notNull(parentIBaseDataObject, "Required: parentIBaseDataObject not null");
+        Validate.notNull(childIBaseDataObject, "Required: childIBaseDataObject not null");
+        Validate.notNull(excludedParametersPattern, "Required: excludedParametersPattern not null");
+        StreamSupport.stream(parentIBaseDataObject.getParameterKeys().spliterator(), false)
```

Comment:
Java 11 allows this to be simplified if we want.
```suggestion
        parentIBaseDataObject.getParameterKeys().stream()
```